### PR TITLE
Disable simd for gcc4.9

### LINKED ISSUE #2017: severe performance issues of SIMD code with GCC 4.9
Something is  bugged in GCC 4.9 or in our SIMD code in relation to GCC 4.9.
Lambda run times with SIMD enabled are on average 10-15% better than without SIMD. This is true for GCC>=5 and clang.
For gcc-4.9 however the run-time **increases** by a factor of 2x to over 10x! I am not sure what that amount of slow-down depends on, it seems to just increase with the amout of SIMD that you are doing (in a non-linear fashion).

I have verified this with gcc-4.9.4 on FreeBSD and gcc-4.9.2 on Debian.

I propose 
* [ ] we deactivate SIMD on gcc-4.9 for now [ for 2.3.2 ?]
* [ ] we debug the problem if possible [ for 2.4.0 ?]

### PR DESCRIPTION
**Pull request by marehr**
Fixes #2017 and current >1000 build-errors on cdash fo gcc 4.9.

Errors:
> /.../Linux-3.16.0-4-amd64_g++-4.9_64/tests/simd/test_simd_vector.h:122:7: error: 'constexpr SimdVectorTestCommon<__vector(32) signed char>::SimdVectorTestCommon()' conflicts with a previous declaration
>
> class SimdVectorTestCommon : public seqan::Test
>       ^

### CODE REVIEW

File: include/seqan/simd.h [hunk 1]
```diff
@@ -86,6 +86,18 @@
     #error SEQAN::SIMD (vector extension) is not supported by msvc and windows intel compiler, try compiling with -DSEQAN_UMESIMD_ENABLED
 #endif
 
+// SIMD operations have severe performance issues on <= gcc4.9
+#if defined(SEQAN_SEQANSIMD_ENABLED) && defined(COMPILER_GCC) && (__GNUC__ <= 4)
+    // TODO(marehr): If we switch to jenkins, filter out these warnings
+    #if !(defined(NDEBUG) || defined(SEQAN_ENABLE_TESTING))
```
Comment (h-2) [inline include/seqan/simd.h]: I would just not emit a warning, it just needlessly clutters the logs. We simply document gcc-4.9 as not supported SIMD and that's it

File: util/cmake/SeqAnSimdUtility.cmake [hunk 2]
```diff
@@ -266,7 +267,12 @@ int main() {
   auto z = x + y;
 
   // gcc 4.9 bug
-  constexpr auto length = LENGTH<int32x8_t>::VALUE;
+  constexpr auto length1 = LENGTH<int32x4_t>::VALUE;
+  constexpr auto length2 = LENGTH<int32x8_t>::VALUE;
+  static_assert(length1 == 4u, \"\");
+  static_assert(length2 == 8u, \"\");
+  std::cout << \"length1: \" << length1 << std::endl;
+  std::cout << \"length2: \" << length2 << std::endl;
```
Comment (h-2) [inline util/cmake/SeqAnSimdUtility.cmake]: How is this related to the above?

Comment (marehr) [reply util/cmake/SeqAnSimdUtility.cmake]: This is the actual cmake code which should have disabled simd on gcc4.9.

But, due to compiler optimization, it did not have an effect.

Comment (h-2) [reply util/cmake/SeqAnSimdUtility.cmake]: Hm, I thought (de)activation of SIMD happens via detection in the library? So we have, both, detection in library and in cmake? This is bound to be error-prone...
